pkg/reconciler/taskrun/resources/taskref.go

```diff
@@ -165,32 +169,34 @@ func GetTaskFunc(ctx context.Context, k8s kubernetes.Interface, tekton clientset
 // resolveTask accepts an impl of remote.Resolver and attempts to
 // fetch a task with given name. An error is returned if the
 // remoteresource doesn't work or the returned data isn't a valid
-// v1beta1.TaskObject.
-func resolveTask(ctx context.Context, resolver remote.Resolver, name string, kind v1beta1.TaskKind, k8s kubernetes.Interface) (v1beta1.TaskObject, *v1beta1.ConfigSource, error) {
+// v1beta1.Task or v1beta1.ClusterTask
+func resolveTask(ctx context.Context, resolver remote.Resolver, name string, kind v1beta1.TaskKind, k8s kubernetes.Interface) (*v1beta1.Task, *v1beta1.ClusterTask, *v1beta1.ConfigSource, error) {
```

Comment:
instead of returning a task and a clusterTask, could we instead convert a clusterTask definition into a Task and just return the Task?

Reply:
I think for the case of a remote-cluster-task, https://github.com/tektoncd/pipeline/blob/main/pkg/reconciler/taskrun/resources/taskref_test.go#L376-L407 the resolveTask and the upstream of its caller GetTaskFunc shall return a ClusterTask at the end?

Reply:
Not sure about those tests, but we just need to fetch the task definition from the cluster, store the task spec in the taskrun status, and execute it. we don't need to keep track of the fact that it's a clustertask. If you take a look at `resolveTask`, only the name and task spec (not the task kind) are used from the results of `getTask`.

Reply:
Thanks @lbernick , I updated the alternative at https://github.com/tektoncd/pipeline/pull/6471, would be happy to close the current PR if #6471 is clearer. 

One concern was that I am not fully sure when we swap all v1beta1 to v1, if it is still ideal handling ClusterTask? Would be more explicit having one more return value in the getTaskFunc while we are also converting it to v1.Task?

Reply:
We can probably convert a v1beta1 ClusterTask spec to a v1 TaskSpec, so I'm not concerned about the proposed solution after swapping to v1.